packages/account-abstraction-auth/src/SafeAuth.ts

```diff
@@ -0,0 +1,92 @@
+import { ethers } from 'ethers'
+import EventEmitter from 'events'
+import EthersAdapter from '@safe-global/safe-ethers-lib'
+import SafeServiceClient from '@safe-global/safe-service-client'
+
+import Web3AuthProvider from './auth-providers/Web3AuthProvider'
+import RPC from './EthereumRPC'
+import {
+  SafeAuthClient,
+  SafeAuthConfig,
+  SafeAuthProviderType,
+  SafeAuthSignInData,
+  SafeAuthEvents
+} from './types'
+
+export default class SafeAuth extends EventEmitter {
+  safeAuthData?: SafeAuthSignInData
+  private authClient?: SafeAuthClient
+  private config: SafeAuthConfig
+
+  constructor(providerType: SafeAuthProviderType, config: SafeAuthConfig) {
+    super()
+
+    this.config = config
+    this.initializeAuthProvider(providerType)
```

Comment:
this method is async but the constructor is not, customers may run into race conditions if they call some methods before the initializer finishes running right?